server/api/api.go

```diff
@@ -1443,6 +1448,250 @@ func (a *API) getWorkspaceUsers(w http.ResponseWriter, r *http.Request) {
+	//   description: Workspace ID
+	//   required: true
+	//   type: string
+	// - name: BlockID
+	//   in: path
+	//   description: Block ID
+	//   required: true
+	//   type: string
+	// - name: subscriberID
+	//   in: path
+	//   description: Subscriber ID
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	ctx := r.Context()
+	session := ctx.Value(sessionContextKey).(*model.Session)
+
+	vars := mux.Vars(r)
+	blockID := vars["blockID"]
+	subscriberID := vars["subscriberID"]
+
+	container, err := a.getContainer(r)
+	if err != nil {
+		a.noContainerErrorResponse(w, r.URL.Path, err)
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "deleteSubscription", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelModify, auditRec)
+	auditRec.AddMeta("block_id", blockID)
+	auditRec.AddMeta("subscriber_id", subscriberID)
+
+	// User can only delete subscriptions for themselves
+	if session.UserID != subscriberID {
+		a.errorResponse(w, r.URL.Path, http.StatusBadRequest, "userID and subscriberID mismatch", nil)
+		return
+	}
+
+	_, err = a.app.DeleteSubscription(*container, blockID, subscriberID)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	a.logger.Debug("DELETE subscription",
+		mlog.String("blockID", blockID),
+		mlog.String("subscriberID", subscriberID),
+	)
+	jsonStringResponse(w, http.StatusOK, "{}")
+
+	auditRec.Success()
+}
+
+func (a *API) handleGetSubscriptions(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation GET /api/v1/workspaces/{workspaceID}/subscriptions/{subscriberID} getSubscriptions
+	//
+	// Gets subscriptions for a user.
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: workspaceID
+	//   in: path
+	//   description: Workspace ID
+	//   required: true
+	//   type: string
+	// - name: subscriberID
+	//   in: path
+	//   description: Subscriber ID
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/User"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	ctx := r.Context()
+	session := ctx.Value(sessionContextKey).(*model.Session)
+
+	vars := mux.Vars(r)
+	subscriberID := vars["subscriberID"]
+
+	container, err := a.getContainer(r)
+	if err != nil {
+		a.noContainerErrorResponse(w, r.URL.Path, err)
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "getSubscriptions", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelRead, auditRec)
+	auditRec.AddMeta("subscriber_id", subscriberID)
+
+	// User can only get subscriptions for themselves
+	if session.UserID != subscriberID {
```

Comment:
If I only allow the session user, makes sense to require the subscriberID? do we expect in the future allow other users (admins for example) to see the subscriptions of other people?

Reply:
There is no current requirement for allowing other users to view your subscriptions.  I guess there is no harm in allowing it.  I'll remove the constraint for reading subscriptions.

Reply:
The code now allows any valid session user to fetch subscription list.  https://github.com/wiggin77/focalboard/commit/4a4a1b5d6afa918a65c8a34d59ad93a7029a3300

Reply:
Sorry, my point here was, If I'm subscribing to something, was there any reason to specify my username in the URL? I'm subscribing, so... I'm the person doing it. Now that you have changed that, and you allow to subscribe other users, you need to have permissions restrictions around that. You only can subscribe to other users if you have some special permission (I think that is getting us in a very weird rabbit hole).

I think the right way to do it, for now, is to remove the userID from the URL, and use the Session.UserID as the subscription user (that means, the subscriber is always the user who call the endpoint).

Reply:
As discussed, the subscriber id will in future be userid, channelid or even teamid.  While the API currently only allows userid, the rest of the plumbing needs a subscriber id field or it will be a larger refactor when we allow it.